ui/src/components/CheckBoxSelectPair.vue

```diff
@@ -126,13 +117,16 @@ export default {
     arrayHasItems (array) {
       return array !== null && array !== undefined && Array.isArray(array) && array.length > 0
     },
-    getSelectInitialValue () {
-      const initialValue = this.selectSource?.filter(x => x.enabled !== false)?.[0]?.id || ''
+    getSelectInitialValue (selectSource) {
+      const initialValue = selectSource?.filter(x => x.enabled !== false)?.[0]?.id || ''
       this.handleSelectChange(initialValue)
       return initialValue
     },
     handleCheckChange (e) {
       this.checked = e.target.checked
+      if (this.checked && !this.selectedOption) {
+        this.selectedOption = this.selectSource[0]?.id || null
+      }
```

Comment:
@utchoang I'm not sure about this as we should probably send signal even when checkbox is unchecked to update the parent form. User may first check and then uncheck the checkbox

Reply:
@shwstppr Currently, selecting or deselecting here both sends the event to the parent form. You can run and check again.

Reply:
And when unchecked, selectedOption is also hidden so I think it doesn't affect